File: mvn-advent-of-code/src/test/java/advent/twenty_fifteen/Day12JSAbacusFrameworkIOTest.java
package advent.twenty_fifteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day12JSAbacusFrameworkIOTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day12JSAbacusFrameworkIO("/twenty_fifteen/Day12-JSAbacusFrameworkIO-input.txt");

        assertEquals(111754, dp.getPart1Answer().intValue());
        assertEquals(65402, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day12JSAbacusFrameworkIO("/twenty_fifteen/Day12-JSAbacusFrameworkIO-example#1.txt");

        assertEquals(18, dp.getPart1Answer().intValue());
        assertEquals(18, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample2() {
        DailyProblem<Integer, Integer> dp = new Day12JSAbacusFrameworkIO("/twenty_fifteen/Day12-JSAbacusFrameworkIO-example#2.txt");

        assertEquals(33, dp.getPart1Answer().intValue());
        assertEquals(16, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_fifteen/Day12JSAbacusFrameworkIOTest.scala
package advent.twenty_fifteen

import org.scalatest.{FlatSpec, Matchers}

class Day12JSAbacusFrameworkIOTest extends FlatSpec with Matchers {
  "2015 Day 12 - Input" should "calculate answers" in {
    val dp = new Day12JSAbacusFrameworkIO("/twenty_fifteen/Day12-JSAbacusFrameworkIO-input.txt")

    dp.part1Answer should be(111754)
    dp.part2Answer should be(65402)
  }

  "2015 Day 12 - Example #1" should "calculate answers" in {
    val dp = new Day12JSAbacusFrameworkIO("/twenty_fifteen/Day12-JSAbacusFrameworkIO-example#1.txt")

    dp.part1Answer should be(18)
    dp.part2Answer should be(18)
  }

  "2015 Day 12 - Example #2" should "calculate answers" in {
    val dp = new Day12JSAbacusFrameworkIO("/twenty_fifteen/Day12-JSAbacusFrameworkIO-example#2.txt")

    dp.part1Answer should be(33)
    dp.part2Answer should be(16)
  }
}
